### PR TITLE
feat: Revamp the website with Jekyll

### PR DESCRIPTION
**Pull request by zusyed**
This PR updates the website to use Jekyll templates.

This addresses https://github.com/tophat/opensource.tophat.com/issues/64

# QA

- You need Ruby to [install Jekyll](https://jekyllrb.com/docs/). I had trouble installing Jekyll with the builtin system ruby so I installed RVM by following [this guide](https://usabilityetc.com/articles/ruby-on-mac-os-x-with-rvm/) but YMMV.
- Once Jekyll is installed, simply run the `bundle exec jekyll serve` command to get the site up and running.
- Verify the site looks as expected.
- Add or remove a project from the [projects.yaml](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_data/projects.yml) file (we can use JSON files instead of YAML files but I think YAML looks cleaner. More info [here](https://jekyllrb.com/docs/datafiles/)) and verify the projects show up as expected.

### Enabling Slack invite forms
- The Slack invite sections are also converted but they are disabled since Slack invite flow is broken at the moment. To enable the invite slack form at the bottom of the index page, uncomment [this](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_includes/join.html#L18-L24).
- To enable the slack invite page, remove `slack.html` from the [config.yml](https://github.com/tophat/opensource.tophat.com/blob/convert-to-jekyll/_config.yml#L5) file. Then restart the server (re-run `bundle exec jekyll serve`) for the changes to take effect.

### CODE REVIEW

**Comment by ju-nc** [comment]
This looks great!

I believe we want to include a `Gemfile`, right? 

Also, when I QA'd, I noticed the project github links say "GitHub", but at the top the link text is "Github"

**Comment by zusyed** [comment]
Yes, `Gemfile` is recommended by [Jekyll](https://jekyllrb.com/docs/step-by-step/10-deployment/). Gemfile is now added.

Also updated the links at the top to say "GitHub".

File: README.md [hunk 1]
```diff
@@ -11,6 +11,10 @@ Top Hats open source portal
 ## I want to Contribute
 The easiest way to get started is to join the [slack workspace](https://opensource.tophat.com/#join-slack) and head to the opensource-portal channel.
 
+### Local Development
+This site uses Jekyll
+ - Install Jekyll by following docs on [their website](https://jekyllrb.com/docs/)
```
Comment (ju-nc) [inline README.md:16]: nb:

could this be updated to run "bundle install"?

**Review by ju-nc — APPROVED** [review]
CR and QA LGTM! One non blocking comment

File: README.md (hunk 1 above)
Comment (zusyed) [reply README.md:16]: Yes, you can use Bundler to install/update Jekyll. Here is the documentation they have for it: https://jekyllrb.com/tutorials/using-jekyll-with-bundler/

I followed the quickstart guide to install Jekyll which is why the quickstart link is in the README. I can add the link to install using bundler too so people are aware that is also an option.

File: Gemfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
+source "https://rubygems.org"
```
Comment (mcataford) [inline Gemfile:1]: **[nb]**

IMO we can take out the boilerplatey comments here.

File: README.md (hunk 1 above)
Comment (zusyed) [reply README.md:16]: Updated the README to include that link. Unfortunately, that removed your approval. Can you approve again?

**Review by ju-nc — APPROVED** [review]
LGTM

**Review by mcataford — COMMENTED** [review]
Looks neat to me!

One thing re:setup/bootstrapping -- any chance we can have a bootstrap script that would set up what we need to build locally? It can be done as a follow-up since it's more devtooley.

File: Gemfile (hunk 2 above)
Comment (zusyed) [reply Gemfile:1]: I agree. Updated to remove those comments.

**Comment by zusyed** [comment]
> Looks neat to me!
> 
> One thing re:setup/bootstrapping -- any chance we can have a bootstrap script that would set up what we need to build locally? It can be done as a follow-up since it's more devtooley.

I can take a look at that. I also think it makes sense to do it as a follow-up.

File: Gemfile (hunk 2 above)
Comment (zusyed) [reply Gemfile:1]: @mcataford Can you approve the PR if you think this PR is good to merge? The change remove the approval.

**Review by mcataford — APPROVED** [review]

File: Gemfile (hunk 2 above)
Comment (mcataford) [reply Gemfile:1]: Done, sorry for the delay, in a meeting. 😅

**Pull request merged**